src/extension.ts

```diff
@@ -0,0 +1,151 @@
+import * as os from 'os';
+import { workspace, ExtensionContext, window, commands } from 'vscode';
+
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+  TransportKind
+} from 'vscode-languageclient/node';
+import { promisify } from 'util';
+import { execFile } from 'child_process';
+
+let client: LanguageClient;
+
+const outputChannel = window.createOutputChannel('Wasp Language Server');
+
+export async function activate(context: ExtensionContext) {
+  console.log('..Extension "vscode-wasp" is now active..');
+
+  const config = workspace.getConfiguration('wasp');
+
+  // Compute path to `waspls` executable, based on settings
+  const userDefinedExecutablePath = config.server.executable;
+  let executablePath = userDefinedExecutablePath === '' ? 'waspls' : userDefinedExecutablePath;
+
+  // Substitute vscode path variables into configuration
+  console.log(`Trying to find the server executable in ${executablePath}`);
+  executablePath = executablePath
+    .replace('${HOME}', os.homedir)
+    .replace('${home}', os.homedir)
+    .replace(/^~/, os.homedir);
+  if (executablePath === '') {
+    window.showErrorMessage(`wasp executable path ${executablePath} is empty, check your configuration`);
+    return;
+  }
+
+  let folders = workspace.workspaceFolders;
+  if (folders) {
+    let folder = folders[0];
+    if (folder) {
+      executablePath = executablePath.replace('${workspaceFolder}', folder.uri.path).replace('${workspaceRoot}', folder.uri.path);
+    }
+  }
+
+  console.log(`Location after path variables subsitution: ${executablePath}`);
+
+  // Check if the path points to a valid waspls executable
+  const executableStatus = await checkExecutable(executablePath);
+
+  if (executableStatus !== 'available') {
+    if (executableStatus === 'timedout') {
+      window.showInformationMessage(`The wasp server process has timed out.`);
+    } else {
+      if (userDefinedExecutablePath === '') {
+        window.showErrorMessage('No `waspls` executable is available in the VSCode PATH.');
+        return;
+      } else {
+        window.showInformationMessage(`The user defined executable path couldn't be run: [${executablePath}].`);
+      }
+    }
+  }
+
```

Comment:
A lot of code for just figuring out the executable path and checking if it is valid!
Maybe we could separate it into a separate function.

Reply:
Refactored into a separate function.